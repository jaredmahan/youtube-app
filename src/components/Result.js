import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Wrapper = styled.div`
  border: 1px solid #ccc;
  border-top-width: 0;
  padding: 10px;
`;

export function Result({ result }) {
  return (
    <Wrapper key={result.id.videoId}>
      <div>
        <a
          target="_blank"
          rel="noopener noreferrer"
          href={`https://www.youtube.com/watch?v=${result.id.videoId}`}
          dangerouslySetInnerHTML={{ __html: result.snippet.title }}
        />
      </div>
      <div>{result.snippet.description}</div>
      <img
        alt="Video Thumbnail"
        src={result.snippet.thumbnails.default.url}
        height={result.snippet.thumbnails.default.height}
        width={result.snippet.thumbnails.default.width}
      />
      {!!result.statistics && !!result.statistics.commentCount && (
        <div>
          {result.statistics.commentCount}
          <span> Comments</span>
        </div>
      )}
    </Wrapper>
  );
}

Result.propTypes = {
  result: PropTypes.object.isRequired
};

export default Result;
