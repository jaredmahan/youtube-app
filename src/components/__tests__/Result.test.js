import React from "react";
import { Result } from "../Result";
import ShallowRenderer from "react-test-renderer/shallow";

describe("<Result>", () => {
  it("renders correctly", () => {
    const renderer = new ShallowRenderer();
    const props = {
      result: {
          id: {
              videoId: "videoId",
          },
          snippet: {
            title: "title",
            description: "description",
            thumbnails: {
                default: {
                    url: "fakeUrl",
                    width: 0,
                    height: 0
                }
            }
          } 
      }
    };
    renderer.render(<Result {...props} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
