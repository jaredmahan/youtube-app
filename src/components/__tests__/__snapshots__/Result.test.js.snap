// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Result> renders correctly 1`] = `
<ForwardRef(styled.div)>
  <div>
    <a
      dangerouslySetInnerHTML={
        Object {
          "__html": "title",
        }
      }
      href="https://www.youtube.com/watch?v=videoId"
      rel="noopener noreferrer"
      target="_blank"
    />
  </div>
  <div>
    description
  </div>
  <img
    alt="Video Thumbnail"
    height={0}
    src="fakeUrl"
    width={0}
  />
</ForwardRef(styled.div)>
`;
